src/interactive/AccordionPanelGroup.jsx

```diff
@@ -11,55 +11,78 @@ class AccordionPanelGroup extends React.Component {
+		this.cloneAccordionPanels = this.cloneAccordionPanels.bind(this);
+	}
 
-		this.clonePanel = this.clonePanel.bind(this);
-		this.setClickedPanel = this.setClickedPanel.bind(this);
+	/**
+	 * @description sets the initial panelStates object based on panel props
+	 */
+	componentWillMount() {
+		// this will be an object in the form [clickId]: isOpen
+		const panelStates = this.accordionPanels.reduce(function(stateObj, panel) {
+			stateObj[panel.props.clickId] = panel.props.isOpen;
+			return stateObj;
+		}, {});
 
-		this.accordionPanels = this.props.accordionPanels.map(this.clonePanel);
+		this.setState({ panelStates });
 	}
 
 	/**
-	 * @param clickedPanel is the `AccordionPanel` component passed in from `_handleToggle`
-	 * @returns {undefined}
-	 *
-	 * Keeps track of the clicked panel in order to support AccordionPanelGroups that only
+	 * @description callback called when individual panel is clicked
+	 * and keeps track of open states in order to support AccordionPanelGroups that only
 	 * have one panel open at a time.
+	 * @param {Integer} clickedPanelId the clickId prop of the `AccordionPanel` component
+	 * @param {Boolean} isOpen whether to open the panel or not (!props.isOpen of the panel)
+	 * @returns {undefined}
 	 */
-	setClickedPanel(clickedPanel) {
-		if (!this.props.multiSelectable) {
-			this.setState({ clickedPanel });
-		}
+	setPanelStates(clickedPanelId, isOpen) {
+		const panelStates = (!this.props.multiSelectable && isOpen) ?
+			Object.keys(this.state.panelStates).reduce(
+				(stateObj, clickId) => {
+					stateObj[clickId] = (parseInt(clickId) === clickedPanelId);
+					return stateObj;
+				}, {}
+			) :
+			{
+				...this.state.panelStates,
+				[clickedPanelId]: isOpen
+			};
+		this.setState({ panelStates });
 	}
 
 	/**
+	 * @description inits a panel with props from the group and individual panel,
+	 * which is then stored as part of accordionPanels array
 	 * @param {Object} accordionPanel - `AccordionPanel` components to clone
-	 * @param {number} i - index of the `AccordionPanel`
-	 * @param {boolean} isOpen - whether the `AccordionPanel` is open or not
-	 * @returns {Array} `AccordionPanel` components with props from `AccordionPanelGroup`
+	 * @param {number} i - index of the `AccordionPanel` used as the key
+	 * @returns {Component} `AccordionPanel` component with props from `AccordionPanelGroup`
 	 */
-	clonePanel(accordionPanel, key, isOpen) {
+	initPanel(panel, index) {
+		const isOpen = panel.props.isOpen || false,
+			clickId = index;
+
 		const panelProps = {
-			key,
 			indicatorAlign: this.props.indicatorAlign,
 			indicatorIcon: this.props.indicatorIcon,
 			indicatorIconActive: this.props.indicatorIconActive,
 			indicatorSwitch: this.props.indicatorSwitch,
-			className: accordionPanel.props.className,
-			setClickedPanel: this.setClickedPanel,
-			isOpen: accordionPanel.props.isOpen
+			className: panel.props.className,
```

Comment:
this line shouldn't be necessary - cloning the panel will retain any original props that are not overridden